blockchain/contracts/HelloWorld.sol

```diff
@@ -47,17 +59,29 @@ contract HelloWorld {
         nftOwnership.dropTime = _dropTime;
         nftOwnerships[nftHash] = nftOwnership;
         availableNFTs.push(nftHash);
-        nftsLength = availableNFTs.length;
+        availableNFTsCount = availableNFTs.length;
     }
 
-    function joinDrop() public {
-        // TODO: purchase limit of 5% of all available NFTs
+    function joinDrop(uint256 _numberOfNFTs) public {
         require(
             !joinedUsers.exists(msg.sender),
             "The address you are calling from has already joined the drop."
         );
-        // We have to make sure that not more users join the drop than we have NFTs available
-        require(joinedUsers.size() <= 3, "You cannot join the drop anymore.");
+        require(
+            reservedNFTsCount != availableNFTs.length,
+            "You cannot join the drop anymore."
+        );
+        require(
+            _numberOfNFTs <= fivePercent,
+            "Sorry, you can't reserve more that 5% of the NFTs."
```

Comment:
Does string interpolation or a similar way exist in solidity to turn the `5` into a variable name?

Reply:
Unfortunately, I didn't find an easy way to do that (yet) (there are no f strings in Solidity like in Python for example). I think for now, whenever we change the variable which is now called percentageLimit we will have to change this message as well.